causalml/inference/tree/models.py

```diff
@@ -920,50 +822,44 @@ def growDecisionTreeFrom(self, X, treatment, y, evaluationFunction, max_depth=10
-            currentScore = evaluationFunction(currentNodeSummary)
-        else:
-            currentScore = evaluationFunction(currentNodeSummary, control_name=self.control_name)
+        currentScore = self.evaluationFunction(currentNodeSummary)
 
         # Prune Stats
         maxAbsDiff = 0
         maxDiff = -1.
-        bestTreatment = self.control_name
-        suboptTreatment = self.control_name
-        maxDiffTreatment = self.control_name
+        bestTreatment = 0       # treatment index for the control group
+        suboptTreatment = 0     # treatment index for the control group
+        maxDiffTreatment = 0    # treatment index for the control group
         maxDiffSign = 0
-        for treatment_group in currentNodeSummary:
-            if treatment_group != self.control_name:
-                diff = (currentNodeSummary[treatment_group][0]
-                        - currentNodeSummary[self.control_name][0])
-                if abs(diff) >= maxAbsDiff:
-                    maxDiffTreatment = treatment_group
-                    maxDiffSign = np.sign(diff)
-                    maxAbsDiff = abs(diff)
-                if diff >= maxDiff:
-                    maxDiff = diff
-                    suboptTreatment = treatment_group
-                    if diff > 0:
-                        bestTreatment = treatment_group
+        p_c, n_c = currentNodeSummary[0]
+        for i_treatment in range(1, self.n_class):
+            p_t, _ = currentNodeSummary[i_treatment]
+            # P(Y|T=t) - P(Y|T=0)
+            diff = p_t - p_c
+            if abs(diff) >= maxAbsDiff:
+                maxDiffTreatment = i_treatment
+                maxDiffSign = np.sign(diff)
+                maxAbsDiff = abs(diff)
+            if diff >= maxDiff:
+                maxDiff = diff
+                suboptTreatment = i_treatment
+                if diff > 0:
+                    bestTreatment = i_treatment
         if maxDiff > 0:
-            pt = currentNodeSummary[bestTreatment][0]
-            nt = currentNodeSummary[bestTreatment][1]
-            pc = currentNodeSummary[self.control_name][0]
-            nc = currentNodeSummary[self.control_name][1]
-            p_value = (1. - stats.norm.cdf((pt - pc) / np.sqrt(pt * (1 - pt) / nt + pc * (1 - pc) / nc))) * 2
+            p_t = currentNodeSummary[bestTreatment][0]
+            n_t = currentNodeSummary[bestTreatment][1]
         else:
-            pt = currentNodeSummary[suboptTreatment][0]
-            nt = currentNodeSummary[suboptTreatment][1]
-            pc = currentNodeSummary[self.control_name][0]
-            nc = currentNodeSummary[self.control_name][1]
-            p_value = (1. - stats.norm.cdf((pc - pt) / np.sqrt(pt * (1 - pt) / nt + pc * (1 - pc) / nc))) * 2
+            p_t = currentNodeSummary[suboptTreatment][0]
+            n_t = currentNodeSummary[suboptTreatment][1]
+
+        p_value = (1. - stats.norm.cdf((p_c - p_t) / np.sqrt(p_t * (1 - p_t) / n_t + p_c * (1 - p_c) / n_c))) * 2
```

Comment:
to confirm, do we need to use `p_t - p_c` when `maxDiff > 0` (if I understand the origin logic correctly) ?

Reply:
I think you're right. @t-tte, could you confirm?

Reply:
Fixed it.